buildSrc/src/main/java/org/elasticsearch/gradle/LoggedExec.java

```diff
@@ -33,6 +34,12 @@ public LoggedExec() {
                             }
                         } catch (UnsupportedEncodingException e) {
                             throw new GradleException("Failed to read exec output", e);
+                        } finally {
+                            try {
+                                output.close();
```

Comment:
I am confused as closing a byte array output stream has no impact? And the backing byte array will be released when the byte array output stream is collected, which it is eligible for when this method exits. What am I missing?

Reply:
The output stream is passed to Gradle above, I suspect the task holds on to a reference and the task itself is not collected in the daemon for reasons that are not clear to me. 
But you're right the close doesn't have any effect so this won't  solve the problem.